providers/providers_manager.go

```diff
@@ -56,22 +86,23 @@ type getProv struct {
 }
 
 // NewProviderManager constructor
-func NewProviderManager(ctx context.Context, local peer.ID, dstore ds.Batching) *ProviderManager {
+func NewProviderManager(ctx context.Context, local peer.ID, dstore ds.Batching, opts ...Option) (*ProviderManager, error) {
 	pm := new(ProviderManager)
 	pm.getprovs = make(chan *getProv)
 	pm.newprovs = make(chan *addProv)
 	pm.dstore = autobatch.NewAutoBatching(dstore, batchBufferSize)
 	cache, err := lru.NewLRU(lruCacheSize, nil)
```

Comment:
hydra is probably the only case where there's a worry about this being expensive, so if this optimization doesn't seem important there, it may not be worth the complexity.
consider checking if the cache is unset after applying options, and only creating the fallback/default one if it hasn't.